lib/sycamore/sycamore/utils/xycut.py

```diff
@@ -0,0 +1,271 @@
+"""
+Sort Elements into reading order based on bboxes using X-Y Cut.
+
+Most useful functions are at the bottom of the file.
+"""
+
+from io import StringIO
+from typing import Generator, Optional
+
+from sycamore.data import Document, Element
+from sycamore.utils.bbox_sort import collect_pages
+
+ElemList = list[Element]
+BeginEndList = list[tuple[float, int, Element]]
+
+XAXIS = 0
```

Comment:
`# The enum values are important bbox[XAXIS], bbox[XAXIS+2] need to be the left and right sides of the bbox.`